### DISCUSSION TITLE
Add warning wave for recursive generic struct

### DISCUSSION THREAD

**Discussion by YairHalberstadt** [dotnet/csharplang#4028] [General] [answered]
There is an existing bug in CoreCLR (not mono or CoreRT) where the following legal code throws a type load exception:

```csharp
static class Program
{
    static void Main()
    {
        var m = new M();
    }
}

struct N<T> { }
struct M { public N<M> E; }
```

The bug is unlikely to be fixed any time soon. See discussion starting at https://github.com/dotnet/runtime/issues/6924#issuecomment-547078135.

I think this would be a good candidate for a warning wave so people don't accidentally write code, which whilst legal, won't actually run in the most popular runtime.

**Comment by 333fred** [comment, marked as answer]
That seems like a fine thing to warn on, but should probably be raised as on issue on Roslyn instead.

**Reply by YairHalberstadt** [reply]
Can you move it please - I wasn't sure where warning waves go

**Reply by 333fred** [reply]
I can't move a discussion to be an issue. :)

**Reply by YairHalberstadt** [reply]
Of course! Opened https://github.com/dotnet/roslyn/issues/48741
